Sources/Prometheus/PrometheusMetricsFactory.swift

```diff
@@ -0,0 +1,132 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CoreMetrics
+
+/// A wrapper around ``PrometheusClient`` to implement the `swift-metrics` `MetricsFactory` protocol
+public struct PrometheusMetricsFactory: Sendable {
+    /// The underlying ``PrometheusClient`` that is used to generate
+    public var client: PrometheusCollectorRegistry
+
+    /// The histogram buckets for a ``TimeHistogram``
+    public var defaultTimeHistogramBuckets: [Duration]
+
+    /// The histogram buckets for a ``TimeHistogram`` per Timer label
+    public var timeHistogramBuckets: [String: [Duration]]
+
+    /// The default histogram buckets for a ``ValueHistogram``. If there is no explicit overwrite
+    /// via ``valueHistogramBuckets``, the buckets provided here will be used for any new
+    /// Swift Metrics Summary type.
+    public var defaultValueHistogramBuckets: [Double]
+
+    public var valueHistogramBuckets: [String: [Double]]
+
+    public var labelAndDimensionSanitizer: @Sendable (_ label: String, _ dimensions: [(String, String)]) -> (String, [(String, String)])
+
+    public init(client: PrometheusCollectorRegistry) {
```

Comment:
Can we get some docs on those three things?

Reply:
Also do I have to create two things now to bootstrap prometheus? Both the `registry` and the `factory`? Couldn't we just default this parameter here to `.init()`?

Reply:
I wonder if we should create a new one, or if we should provide a `.default` singleton. I think I'd be more in favor of the `.default`, wdyt?

Reply:
I don't see a big benefit on a default singleton here and we avoided introducing singletons in AHC because we want to discuss this in the SSWG in a broader scope. So I would say we shouldn't start introducing it here. Just `.init()` default seems fine to me.